### PR TITLE
Add image and clipboard paste support via GDClip

### PR DESCRIPTION
**Pull request by hansemro**
This pull request introduces image support and partial clipboard support to resolve the main objective of #100. Image resizing or transform is not yet supported.

Notable changes:
- Images are drawn as ImageStroke instances, which can check for visibility and work with existing tools with minimal changes.
- Images are pasted using GDClip cross-platform GDNative library. 
  - Tested on macOS (Big Sur), Windows, Linux with Godot 3.4.x.
  - macOS 10.9 and newer should be supported. Please report bugs if there are any.
- All SelectionTool features work (including undo/redo actions)
  - Images are selected when intersecting selection rectangle. Let me know if you want the images to be selected only if enclosed entirely by selection rectangle.
  - Note: Images copied with SelectionTool do not save to clipboard.
- New save format that includes images. This breaks compatibility with older versions lacking image support.
  - Images will be compressed to reduce file size and improve save/load performance.
- Images are now embedded in the SVG export as base64-encoded PNGs.

Additional Tasks:
- [x] Update manual
- [x] Update README.md
- [x] Update CI to build with GDClip libraries
- [ ] ~Disable image paste feature if gdclip library cannot be found~

Pre-Release builds can be found here: https://github.com/hansemro/Lorien/releases

### CODE REVIEW

**Comment by hansemro** [comment]
Sorry for the delay, but I finally got around to setup CI infrastructure for GDClip. Let me know if there is a change you want me to make.

File: lorien/gdclip.gdnlib [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[general]
```
Comment (hansemro) [inline lorien/gdclip.gdnlib:1]: What would be a good location for `gdclip.gdnlib` and `gdclip.gdns`?

Comment (hansemro) [reply lorien/gdclip.gdnlib]: Moved GDClip-related files under `lorien/InfiniteCanvas/` where it is used.

File: lorien/ProjectManager/Serializer.gd [hunk 2]
```diff
@@ -105,14 +120,39 @@ static func load_project(project: Project) -> void:
 					var pressure := file.get_8()
 					brush_stroke.points.append(Vector2(x, y))
 					brush_stroke.pressures.append(pressure)
-				
+				project.strokes.append(brush_stroke)
```
Comment (hansemro) [inline lorien/ProjectManager/Serializer.gd:123]: Will remove duplicate line.

File: lorien/InfiniteCanvas/Tools/SelectionTool.gd [hunk 3]
```diff
@@ -148,29 +151,37 @@ func _paste_strokes(strokes: Array) -> void:
...
 	var duplicates := []
 	for stroke in strokes:
-		var dup := _duplicate_stroke(stroke, offset)
+		var dup = _duplicate_stroke(stroke, offset)
 		dup.add_to_group(GROUP_SELECTED_STROKES)
 		dup.modulate = Config.DEFAULT_SELECTION_COLOR
 		duplicates.append(dup)
 	
 	_canvas.add_strokes(duplicates)
-	print("Pasted %d strokes (offset: %s)" % [strokes.size(), offset])
```
Comment (hansemro) [inline lorien/InfiniteCanvas/Tools/SelectionTool.gd:157]: Will re-add deleted line.

**Comment by hansemro** [comment]
I had forgotten to update interaction with (super) EraserTool on ImageStroke. Should be fixed now.

**Comment by MrApplejuice** [comment]
Quick question: I read "SvgExporter" - is that to actually make it possible to copy-paste strokes from Lorien to, for example, Inkscape? That was a feature I was about to suggest after finishing my keybindings-PR. If it is already done, however, that changes stuff, and will make me super-duper happy, of course :-)

**Comment by hansemro** [comment]
@MrApplejuice SVG export support is already in v0.5.0 and accessible from hamburger menu -> Export.

fad4ea9 just embeds images in the svg file.

**Comment by MrApplejuice** [comment]
Ah thanks, for clarification! Maybe I will propose a copy-pasta in SVG format later then. I would find it super-handy to be able to easily exchange stuff among different types of vector-graphics editors. I find Lorien way more convenient for "quick doodles" but then sometimes the doodles outgrow Lorien and I need to go to something more advanced. The indirect way of Export is a bit cumerbsome for fast progress.

That said: Nice PR. Pasting images will make the sketching-process a lot easier again! :-)

**Comment by hansemro** [comment]
Latest changes include moving paste action to a new thread so that main thread is not blocked.

**Pull request closed without merging**